### DISCUSSION TITLE
branches

### DISCUSSION THREAD

**Discussion by zophiale** [community/community#22553] [Other Feature Feedback, Questions, & Ideas]
<p>how can i rename/delete branches?</p>

**Comment by Wabri** [comment, marked as answer]
<p>To rename a branch you simply need to create a new one starting from the branch you want to rename. And when you create this new branch you can simply delete the old branch.</p>
<p>An example: I want to rename the branch called “old” to “new”. To do this with the command line you need to be in the “old” branch, so if you are not in this you need to switch to it:</p>
<pre><code>git checkout old
</code></pre>
<p>Now you have to create a new branch and call it “new”:</p>
<pre><code>git checkout -b new
</code></pre>
<p>The “old” branch is useless than you can delete that:</p>
<pre><code>git branch -D old
</code></pre>
<p>if you have “old” also in remote repository you have to delete it with:</p>
<pre><code>git push origin --delete old
</code></pre>
<p>Where origin is the remote repository.</p>
<p>Hope to solve your problem :slightly_smiling_face: </p>
<p>-Gabriele-</p>
